libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts

```diff
@@ -372,8 +373,16 @@ export class WorkspaceInstallManager extends BaseInstallManager {
     }
 
     // Check if overrides and globalOverrides are the same
+    const repoPnpmConfiguration: PnpmOptionsConfiguration = this.rushConfiguration.pnpmOptions;
+    const subspacePnpmConfiguration: PnpmOptionsConfiguration | undefined = subspace.getPnpmOptions();
+
+    const mergedGlobalOverrides: Record<string, string> = {
+      ...(repoPnpmConfiguration.globalOverrides ?? {}),
+      ...(subspacePnpmConfiguration?.globalOverrides ?? {})
+    };
```

Comment:
```suggestion
    const mergedGlobalOverrides: Record<string, string> = {
      ...repoPnpmConfiguration.globalOverrides,
      ...subspacePnpmConfiguration?.globalOverrides
    };
```

The spread operator when assigning to an object handles `undefined`.